test/helpers/privateMocks.js

```diff
@@ -11,7 +11,7 @@ function baseThey(msg, vals, spec, itFn) {
   var valsIsArray = angular.isArray(vals);
 
   angular.forEach(vals, function(val, key) {
-    var m = msg.replace(/\$prop/g, angular.toJson(valsIsArray ? val : key));
+    var m = msg.split('$prop').join(angular.toJson(valsIsArray ? val : key));
```

Comment:
It's described in the commit message of the [first commit][1].

Basically, if your replacement (`val` or `key`) contains `$&`, it will be replaced with `$prop`.
Eg. `'Testing $prop'.replace(/\$prop/g, 'string with $&') === 'Testing string with $prop'`.

[1]: https://github.com/gkalpak/angular.js/commit/031303d3271905faf265470a6effe12ee7eebf6e